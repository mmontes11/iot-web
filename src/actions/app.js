import { TOGGLE_HAMBURGER_MENU, TOGGLE_MAP_DIALOG, SET_SHOW_ERROR } from "constants/actionTypes/app";

export const toggleHamburgerMenu = () => dispatch => {
  dispatch({ type: TOGGLE_HAMBURGER_MENU });
};

export const toggleMapDialog = () => dispatch => {
  dispatch({ type: TOGGLE_MAP_DIALOG });
};

export const setShowError = showError => dispatch => {
  dispatch({ type: SET_SHOW_ERROR, showError });
};
