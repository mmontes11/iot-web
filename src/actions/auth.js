import { IS_AUTH } from "constants/actionTypes/auth";
import iotClient from "lib/iotClient";

export const isAuth = () => dispatch => {
  iotClient.authService
    .isAuth()
    .then(isAuthVal => dispatch({ type: IS_AUTH, isAuth: isAuthVal }))
    .catch(() => dispatch({ type: IS_AUTH, isAuth: false }));
};
