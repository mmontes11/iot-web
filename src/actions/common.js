import { RESET } from "constants/actionTypes/common";

export const reset = () => dispatch => {
  dispatch({ type: RESET });
};
