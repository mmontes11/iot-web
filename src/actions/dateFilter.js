import {
  DATE_FILTER_SELECTOR_TOGGLE,
  DATE_FILTER_TIME_PERIOD_SELECT,
  DATE_FILTER_TIME_PERIOD_UPDATED,
  DATE_FILTER_TIME_PERIODS_REQUEST,
  DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS,
  DATE_FILTER_TIME_PERIODS_REQUEST_ERROR,
  DATE_FILTER_TIME_PERIODS_UPDATED,
  DATE_FILTER_START_DATE_UPDATED,
  DATE_FILTER_END_DATE_UPDATED,
} from "constants/actionTypes/dateFilter";
import iotClient from "lib/iotClient";

export const toggleDateFilterType = () => dispatch => {
  dispatch({ type: DATE_FILTER_SELECTOR_TOGGLE });
};

export const selectTimePeriod = isActive => dispatch => {
  dispatch({ type: DATE_FILTER_TIME_PERIOD_SELECT });
  if (!isActive) {
    dispatch({ type: DATE_FILTER_TIME_PERIODS_REQUEST });
    iotClient.timePeriodsService
      .getSupportedTimePeriods()
      .then(({ body: { timePeriods } }) => {
        dispatch({ type: DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS });
        dispatch({ type: DATE_FILTER_TIME_PERIODS_UPDATED, timePeriods });
      })
      .catch(error => {
        dispatch({ type: DATE_FILTER_TIME_PERIODS_REQUEST_ERROR, statusCode: error.statusCode, error });
      });
  }
};

export const updateTimePeriod = updatedTimePeriod => dispatch => {
  dispatch({ type: DATE_FILTER_TIME_PERIOD_UPDATED, updatedTimePeriod });
};

export const updateStartDate = updatedStartDate => dispatch => {
  dispatch({ type: DATE_FILTER_START_DATE_UPDATED, updatedStartDate });
};

export const updateEndDate = updatedEndDate => dispatch => {
  dispatch({ type: DATE_FILTER_END_DATE_UPDATED, updatedEndDate });
};
