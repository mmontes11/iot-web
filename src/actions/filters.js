import { FILTER_TYPE_SELECT, ADD_FILTER_TYPE } from "constants/actionTypes/filters";

export const selectFilterType = () => dispatch => {
  dispatch({ type: FILTER_TYPE_SELECT });
};

export const addFilterType = addedFilterType => dispatch => {
  dispatch({ type: ADD_FILTER_TYPE, addedFilterType });
};
