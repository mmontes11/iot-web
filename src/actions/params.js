import {
  TYPE_SELECT,
  TYPE_UPDATED,
  OBSERVATIONS_REQUEST,
  OBSERVATIONS_REQUEST_SUCCESS,
  OBSERVATIONS_REQUEST_ERROR,
  OBSERVATIONS_UPDATED,
} from "constants/actionTypes/params";
import { EVENT_TYPE, MEASUREMENT_TYPE } from "constants/observationTypes";
import iotClient from "lib/iotClient";

const requestTypes = (type, onStart, onSuccess, onError) => {
  if (type === EVENT_TYPE) {
    onStart();
    iotClient.eventService
      .getTypes()
      .then(res => onSuccess(res))
      .catch(err => onError(err));
  } else if (type === MEASUREMENT_TYPE) {
    onStart();
    iotClient.measurementService
      .getTypes()
      .then(res => onSuccess(res))
      .catch(err => onError(err));
  }
};

export const selectType = () => dispatch => {
  dispatch({ type: TYPE_SELECT });
};

export const updateType = updatedType => dispatch => {
  dispatch({ type: TYPE_UPDATED, updatedType });
  requestTypes(
    updatedType,
    () => dispatch({ type: OBSERVATIONS_REQUEST }),
    res => {
      dispatch({ type: OBSERVATIONS_REQUEST_SUCCESS, statusCode: res.statusCode, error: null });
      dispatch({ type: OBSERVATIONS_UPDATED, things: res.body.types });
    },
    error => {
      dispatch({ type: OBSERVATIONS_REQUEST_ERROR, statusCode: error.statusCode, error });
    },
  );
};
