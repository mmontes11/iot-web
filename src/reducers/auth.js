import { IS_AUTH, SET_PASSWORD, SET_USERNAME, SET_SHOW_ERROR } from "constants/actionTypes/auth";
import { RESET } from "constants/actionTypes/common";

const initialState = {
  isAuth: false,
  username: null,
  password: null,
  showError: true,
};

export default (state = initialState, { type, isAuth, username, password, showError }) => {
  switch (type) {
    case IS_AUTH:
      return { ...initialState, isAuth };
    case SET_USERNAME:
      return { ...state, username };
    case SET_PASSWORD:
      return { ...state, password };
    case SET_SHOW_ERROR:
      return { ...state, showError };
    case RESET:
      return { ...initialState, isAuth: state.isAuth };
    default:
      return state;
  }
};
