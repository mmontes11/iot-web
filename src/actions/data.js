import { DATA_UPDATED, DATA_REQUEST, DATA_REQUEST_SUCCESS, DATA_REQUEST_ERROR } from "constants/actionTypes/data";
import { EVENT_TYPE, MEASUREMENT_TYPE } from "constants/observationTypes";
import iotClient from "lib/iotClient";
import { THING_FILTER_TYPE, DATE_FILTER_TYPE } from "constants/filterTypes";
import { RESET } from "constants/actionTypes/common";
import * as fromState from "reducers";
import { TYPE, OBSERVATION } from "constants/params";

const getParams = (filterItems, observation, thing, timePeriod, startDate, endDate) => {
  let params = {
    type: observation,
  };
  if (filterItems.includes(THING_FILTER_TYPE) && thing !== null) {
    params = {
      ...params,
      thing,
    };
  }
  if (filterItems.includes(DATE_FILTER_TYPE)) {
    if (timePeriod != null) {
      params = {
        ...params,
        timePeriod,
      };
    } else {
      if (startDate !== null) {
        params = {
          ...params,
          startDate: startDate.toISOString(),
        };
      }
      if (endDate != null) {
        params = {
          ...params,
          endDate: endDate.toISOString(),
        };
      }
    }
  }
  return params;
};

const requestStats = (
  filterItems,
  type,
  observation,
  thing,
  timePeriod,
  startDate,
  endDate,
  onStart,
  onSuccess,
  onError,
) => {
  const params = getParams(filterItems, observation, thing, timePeriod, startDate, endDate);
  if (type === EVENT_TYPE) {
    onStart();
    iotClient.eventService
      .getStats(params)
      .then(res => onSuccess(res))
      .catch(err => onError(err));
  } else if (type === MEASUREMENT_TYPE) {
    onStart();
    iotClient.measurementService
      .getStats(params)
      .then(res => onSuccess(res))
      .catch(err => onError(err));
  } else {
    onError(new Error("Invalid type"));
  }
};

export const getStats = () => (dispatch, getState) => {
  const state = getState();
  const filterItems = state.filters.items;
  const type = fromState.getParam(state, TYPE).selectedItem;
  const observation = fromState.getParam(state, OBSERVATION).selectedItem;
  const thing = state.filters.thingFilter.selectedItem;
  const {
    timePeriod: { selectedItem: selectedTimePeriod },
    custom: { startDate, endDate },
  } = state.filters.dateFilter;
  if (!type || !observation) {
    return;
  }
  requestStats(
    filterItems,
    type,
    observation,
    thing,
    selectedTimePeriod,
    startDate,
    endDate,
    () => dispatch({ type: DATA_REQUEST }),
    res => {
      dispatch({ type: DATA_REQUEST_SUCCESS });
      dispatch({ type: DATA_UPDATED, items: res.body.stats });
    },
    () => {
      dispatch({ type: DATA_REQUEST_ERROR });
      dispatch({ type: RESET, preserveError: true });
    },
  );
};
