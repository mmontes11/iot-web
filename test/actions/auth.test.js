import { isAuth } from "actions/auth";

describe("actions/auth", () => {
  describe("isAuth", () => {
    it("dispatches a isAuth action after success", () => {
      const thunk = isAuth();
      const dispatch = jest.fn();
      thunk(dispatch);
      expect(thunk).toBeAThunk();
      expect(thunk).toMatchSnapshot();
    });
  });
});
