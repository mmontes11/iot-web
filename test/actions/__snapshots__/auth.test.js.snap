// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/auth isAuth dispatches a isAuth action after success 1`] = `[Function]`;
